### PR TITLE
Support npm-style #ref suffix; fixes issue 16

### PR DESCRIPTION
**Pull request by dankegel**
Also save one call to re.search().

### CODE REVIEW

**Comment by dankegel** [comment]
It's an open question whether the last commit should be included.

**Comment by retr0h** [comment]
Why would we be parsing npm urls?

File: test/conftest.py [hunk 1]
```diff
@@ -103,6 +109,34 @@ def git_urls():
...
+            'href': 'user@example.com:owner/repo',
+            'resource': 'example.com',
+            'user': 'user',
+            'port': None,
+            'name': 'repo',
+            'owner': 'owner',
+            'ref': None,
+        },
+        # npm and others allow specifying a branch or commit.  See
+        # https://docs.npmjs.com/files/package.json#git-urls-as-dependencies
```
Comment (retr0h) [inline test/conftest.py]: Why would this ever be used? `https://docs.npmjs.com/files/package.json#git-urls-as-dependencies`?  Thats an NPM url.

Comment (dankegel) [reply test/conftest.py]: That's just an example of one project that uses the convention.  yarn uses it, too.

I've been using it in my scripts to configure buildbot, and I find it useful.

Comment (dankegel) [reply test/conftest.py]: That said, it's not that much code to do without this feature.

Comment (retr0h) [reply test/conftest.py]: Sure, but npm isnt a git url neither is yarn urls.  I have not seen a git url having this format so I would have a hard time accepting a PR like this bc it's outside the scope of the project.

Comment (dankegel) [reply test/conftest.py]: I think it's a useful convention even outside of npm, but it's easy to implement as a tiny wrapper around git-url-parse, so closing.

**Pull request closed without merging**